### PR TITLE
boring: fix Rust 1.72 Clippy lints

### PR DESCRIPTION
**Pull request by hawkw**
Updating to Rust 1.72 results in new Clippy lints showing up for the
`boring` crate. These lints are currently breaking the CI build.

This branch contains two commits which each fix a different Clippy lint.
See the individual commit messages for details:

* 045f02f4 boring: allow `clippy::bad_bit_mask` lints
* 0fa986de boring: fix `clippy::unnecessary_cast` warnings

### CODE REVIEW

File: boring/src/ssl/mod.rs [hunk 1]
```diff
@@ -57,6 +57,20 @@
...
+// literal `0`. However, `& 0` shows up in the code generated by the `bitflags!`
+// macro --- presumably, this is because it's simpler to have the macro generate
+// expressions like `bits & 1 << 2`, `bits & 1 << 1`, and finally `bits & 1 <<
+// 0` or similar, rather than special-casing the 0 case. Thus, allow this lint
+// in this module.
+//
+// Unfortunately, we can't put the `allow` attribute on the `bitflags!` type
+// itself, because the macro doesn't accept arbitrary attributes on the
+// generated type. So, we have to allow this for the whole module.
+#![allow(clippy::bad_bit_mask)]
```
Comment (inikulin) [inline boring/src/ssl/mod.rs:72]: Just fixed it in context of https://github.com/cloudflare/boring/pull/160, the fix is to update bitflags to ^2.0

**Review by inikulin — CHANGES_REQUESTED** [review]

File: boring/src/ssl/mod.rs (hunk 1 above)
Comment (inikulin) [reply boring/src/ssl/mod.rs:72]: I think we can just merge #160 and just rebase your previous PR on top of it

Comment (hawkw) [reply boring/src/ssl/mod.rs:72]: sounds good to me --- happy to drop that part of the PR!

Comment (hawkw) [reply boring/src/ssl/mod.rs:72]: @inikulin let me know when I should rebase?

Comment (inikulin) [reply boring/src/ssl/mod.rs:72]: #160 was merged

**Pull request closed without merging**